### PR TITLE
[Torch-MLIR] support for asr

### PR DESCRIPTION
**Pull request by zzpmiracle**

### CODE REVIEW

File: pytorch_blade/pytorch_blade/compiler/jit/torch/shape_analysis.cpp [hunk 1]
```diff
@@ -2318,6 +2317,28 @@ class ShapePropagator : public PropertyPropBase {
...
+          return false;
+
+        std::vector<c10::ShapeSymbol> new_sizes = sizesOptional.value();
+        int64_t input_rank = new_sizes.size();
+        int64_t dim =
+            at::maybe_wrap_dim(dimOptional.value(), input_rank, false);
+
+        if (new_sizes[dim].is_static()) {
+          new_sizes[dim] =
+              ShapeSymbol::fromStaticSize(new_sizes[dim].static_size() / 2);
```
Comment (tanyokwok) [inline pytorch_blade/pytorch_blade/compiler/jit/torch/shape_analysis.cpp:2334]: Nit: Add a file check test

File: pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscDecomposeComplexOps.cpp [hunk 2]
```diff
@@ -287,8 +287,84 @@ LogicalResult ConvertAtenOp<OperatorOp>::matchAndRewrite(
 
     rewriter.replaceOpWithNewOp<AtenMulScalarOp>(op, outTy, addVal, floatScale);
     return success();
+  } else if ("aten.conv1d" == name) {
```
Comment (qiuxiafei) [inline pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscDecomposeComplexOps.cpp:290]: why does conv1d fall here? torch-mlir doesn't have a definition ?

File: pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscDecomposeComplexOps.cpp [hunk 3]
```diff
@@ -287,8 +287,84 @@ LogicalResult ConvertAtenOp<OperatorOp>::matchAndRewrite(
...
+  } else if ("aten.glu" == name) {
+    auto inputTy = op.getOperand(0).getType().cast<BaseTensorType>();
+    int64_t inputRank = inputTy.getSizes().size();
+    Value dim = op.getOperand(1);
+    int64_t dimInt;
+    if (!matchPattern(dim, m_TorchConstantInt(&dimInt)))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: dim must be a constant");
+    if (dimInt < 0)
+      dimInt += inputRank;
```
Comment (tanyokwok) [inline pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscDecomposeComplexOps.cpp:328]: Nit: dimInt = toPositiveDim(dimInt, inputRank);

File: pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscDecomposeComplexOps.cpp [hunk 4]
```diff
@@ -519,7 +595,9 @@ class DiscDecomposeComplexOpsPass
           "aten.narrow.Tensor",
           "aten.selu",
           "aten.selu_",
-      };
+          "aten.conv1d",
```
Comment (tanyokwok) [inline pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscDecomposeComplexOps.cpp:598]: Nit: add an e2e unittest for `aten.conv1d` under `tests/disc/ops`

File: pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscDecomposeComplexOps.cpp (hunk 2 above)
Comment (zzpmiracle) [reply pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscDecomposeComplexOps.cpp:290]: Yes...

File: pytorch_blade/pytorch_blade/compiler/jit/torch/shape_analysis.cpp [hunk 5]
```diff
@@ -771,11 +765,7 @@ class ShapePropagator : public PropertyPropBase {
       return;
```
Comment (tanyokwok) [inline pytorch_blade/pytorch_blade/compiler/jit/torch/shape_analysis.cpp:765]: How about `return setUnshapedType(node);` here?

**Review by tanyokwok — APPROVED** [review]
LGTM

**Pull request merged**